<?xml version="1.0"?>
<!-- j2s7s300 refers to jaco v2 7DOF spherical 3fingers -->
<root xmlns:xacro="http://ros.org/wiki/xacro">
	<xacro:include filename="$(find robot_grasp)/models/table/table_properties.xacro" />
	<gazebo reference="table_base">
		<material>Gazebo/FlatBlack</material>
	</gazebo>
	<gazebo reference="table_top">
		<material>Gazebo/White</material>
	</gazebo>
	
	<link name="table_base">
		<inertial>
			<origin xyz="0.018511509179121 0.23251932391586 1.175991980646075" rpy="0 0 0" />
			<mass value="1265032.07764878" />
			<inertia ixx="169157.982834144" ixy="-162908.934773947E-14" ixz="-974479.440236252E-15" iyy="323322.704330012" iyz="-184123.364573924E-14" izz="259218.197995528" />
		</inertial>
		<visual>
			<origin xyz="0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh filename="$(find robot_grasp)/models/table/meshes/table_base.STL" />
			</geometry>
			<material name="">
				<color rgba="0.192156862745098 0.203921568627451 0.203921568627451 1" />
			</material>
		</visual>
		<collision>
			<origin xyz="0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh filename="$(find robot_grasp)/models/table/meshes/table_base.STL" />
			</geometry>
		</collision>
	</link>
	
	<link name="table_top">
		<inertial>
			<origin xyz="2.18575157973078E-16 0.0165 -1.11022302462516E-16" rpy="0 0 0" />
			<mass value="2772000" />
			<inertia ixx="148091.559" ixy="-470197.740328915E-35" ixz="133744.867033177E-16" iyy="402517.5" iyz="0" izz="254929.059" />
		</inertial>
		<visual>
			<origin xyz="0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh filename="$(find robot_grasp)/models/table/meshes/table_top.STL" />
			</geometry>
			<material name="">
				<color rgba="1 1 1 1" />
			</material>
		</visual>
		<visual>
			<origin xyz="0 ${t_thickness+rim_height/2} ${-t_height/2+rim_thickness/2}" rpy="0 0 0" />
			<geometry>
				<box size="${t_width} ${rim_height} ${rim_thickness}"/>
			</geometry>
			<material name="">
				<color rgba="0.8 0.8 1 1" />
			</material>
		</visual>
		<visual>
			<origin xyz="${t_width/2-rim_thickness/2} ${t_thickness+rim_height/2} 0" rpy="0 0 0" />
			<geometry>
				<box size="${rim_thickness} ${rim_height} ${t_height}"/>
			</geometry>
			<material name="">
				<color rgba="0.8 0.8 1 1" />
			</material>
		</visual>

		<collision>
			<origin xyz="0 ${t_thickness+rim_height/2} ${-t_height/2+rim_thickness/2}" rpy="0 0 0" />
			<geometry>
				<box size="${t_width} ${rim_height} ${rim_thickness}"/>
			</geometry>
		</collision>
		<collision>
			<origin xyz="${t_width/2-rim_thickness/2} ${t_thickness+rim_height/2} 0" rpy="0 0 0" />
			<geometry>
				<box size="${rim_thickness} ${rim_height} ${t_height}"/>
			</geometry>
		</collision> 
		<collision>
			<origin xyz="0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh filename="$(find robot_grasp)/models/table/meshes/table_top.STL" />
			</geometry>
		</collision>
	</link>
<!-- Added   -->

	<joint name="j1" type="fixed">
		<origin xyz="0.0185115091791201 -0.175991980646074 0.397575968372377" rpy="1.5707963267949 0 0" />
		<parent link="table_base" />
		<child link="table_top" />
		<axis xyz="0 0 0" />
	</joint>


	<!-- Add some handy co-ordinate frames -->

	<link name="tabletop_ontop"/>
	<joint name="table_tabletop_totop" type="fixed">
		<origin xyz="0.0 0.033 0" rpy="-1.57079632679 0 0" />
		<parent link="table_top" />
		<child link="tabletop_ontop" />
		<axis xyz="0 0 0" />
	</joint>

	<link name="tabletop_corner"/>
	<joint name="set_tabletop_relto_corner" type="fixed">
		<origin xyz="0.42 0.3 0" rpy="0 0 -0.76" />
		<parent link="tabletop_ontop"/>
		<child link="tabletop_corner"/>
	</joint>

        <!-- Grasp Objects -->
	<gazebo reference="basket">
          <mu1>100.0</mu1>
          <mu2>100.0</mu2>
          <minDepth>0.01</minDepth>
          <material>Gazebo/Green</material>
	</gazebo>
	<gazebo reference="basket2">
          <mu1>100.0</mu1>
          <mu2>100.0</mu2>
          <minDepth>0.01</minDepth>
          <material>Gazebo/Blue</material>
	</gazebo>
	<gazebo reference="kinect_support">
		<material>Gazebo/Black</material>
	</gazebo>

	<link name="basket">
		<inertial>
			<origin xyz="0 0 0" rpy="0.2 0.3 0.4" />
			<mass value="0.1" />
			<inertia ixx="0.02" ixy="0.02" ixz="0.02" iyy="0.02" iyz="0.02" izz="0.02" />
		</inertial>
		<visual>
			<origin xyz="0 0 -0.1" rpy="0 0 0" />
			<geometry>
                                <mesh scale="0.0013 0.0013 0.0013" filename="$(find robot_grasp)/models/grasp_object/basket.dae" />
			</geometry>
		</visual>
		<collision>
			<origin xyz="0 0 -0.1" rpy="0 0 0" />
			<geometry>
                                <mesh scale="0.0013 0.0013 0.0013" filename="$(find robot_grasp)/models/grasp_object/basket.dae" />
			</geometry>
		</collision>
	</link>

	<link name="basket2">
		<inertial>
			<origin xyz="0 0 0" rpy="0.2 0.3 0.4" />
			<mass value="0.1" />
			<inertia ixx="0.02" ixy="0.02" ixz="0.02" iyy="0.02" iyz="0.02" izz="0.02" />
		</inertial>
		<visual>
			<origin xyz="0 0 -0.1" rpy="0 0 0" />
			<geometry>
                                <mesh scale="0.001 0.001 0.001" filename="$(find robot_grasp)/models/grasp_object/basket.dae" />
			</geometry>
		</visual>
		<collision>
			<origin xyz="0 0 -0.1" rpy="0 0 0" />
			<geometry>
                                <mesh scale="0.0008 0.0008 0.0008" filename="$(find robot_grasp)/models/grasp_object/basket.dae" />
			</geometry>
		</collision>
	</link>

	<link name="kinect_support">
		<inertial>
			<origin xyz="0 0 0" rpy="0.2 0.3 0.4" />
			<mass value="0.1" />
			<inertia ixx="0.02" ixy="0.02" ixz="0.02" iyy="0.02" iyz="0.02" izz="0.02" />
		</inertial>
		<visual>
			<origin xyz="0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh scale="0.001 0.001 0.001" filename="$(find robot_grasp)/models/table/meshes/kinect_support.dae" />
			</geometry>
		</visual>
		<collision>
			<origin xyz="0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh scale="0.001 0.001 0.001" filename="$(find robot_grasp)/models/table/meshes/kinect_support.dae" />
			</geometry>
		</collision>
	</link>


	<joint name="table_basket" type="fixed">
                <origin xyz="0.25 0.065 0.1" rpy="0 0 0" />  <!-- relative to world coordinator: y z x 0.25 0.06 0.1-->
		<parent link="table_top" />
		<child link="basket" />
		<axis xyz="0 0 0" />
	</joint>

	<joint name="table_basket2" type="fixed">
                <origin xyz="-0.25 0.06 0.1" rpy="0 0 0" />
		<parent link="table_top" />
		<child link="basket2" />
		<axis xyz="0 0 0" />
	</joint>

	<joint name="table_kinect_support" type="fixed">
                <origin xyz="0.1 0.19 0.25" rpy="0 -1.5707963267949 0" />  <!-- relative to world coordinator: x z -y -->
		<parent link="table_top" /> 
		<child link="kinect_support" />
		<axis xyz="0 0 0" />
	</joint>
    <link name="camera_link">
            <collision>
              <origin xyz="-0.016 0 0" rpy="0 0 0"/>
              <geometry>
                <box size="0.036 0.18 0.04"/>
              </geometry>
            </collision>
            <visual>
              <origin xyz="0 0 0" rpy="0 0 0"/>
              <geometry>
                <mesh filename="package://robot_grasp/models/kinect/meshes/kinect.dae"/>
              </geometry>
            </visual>
            <inertial>
              <mass value="1.0" />
              <origin xyz="0 0 0" rpy="0 0 0"/>
              <inertia ixx="5.8083e-4" ixy="0" ixz="0" iyy="3.0833e-5" iyz="0" izz="5.9083e-4" />
            </inertial>
         </link>

        <joint name="camera_joint" type="fixed">
            <origin xyz="-0.25 0.48 0" rpy="1.5707963267949 0 -1.5707963267949" /><!--"-0.278 0.41 0" -->
            <parent link="kinect_support" />
            <child link="camera_link" />
            <axis xyz="0 1 0"/>
        </joint>

        <link name="camera_rgb_frame" />
          <joint name="camera_rgb_joint" type="fixed">
            <origin rpy="0 0 0" xyz="0 0.022 0" />
            <parent link="camera_link" />
            <child link="camera_rgb_frame" />
          </joint>
          <link name="camera_rgb_optical_frame" />
          <joint name="camera_rgb_optical_joint" type="fixed">
            <origin rpy="-1.57079632679 0 -1.57079632679" xyz="0 0 0" />
            <parent link="camera_rgb_frame" />
            <child link="camera_rgb_optical_frame" />
          </joint>
          <link name="camera_depth_frame" />
          <joint name="camera_depth_joint" type="fixed">
            <origin rpy="0 0 0" xyz="0 0.049 0" />
            <parent link="camera_link" />
            <child link="camera_depth_frame" />
          </joint>
          <link name="camera_depth_optical_frame" />
          <joint name="camera_depth_optical_joint" type="fixed">
            <origin rpy="-1.57079632679 0 -1.57079632679" xyz="0 0 0" />
            <parent link="camera_depth_frame" />
            <child link="camera_depth_optical_frame" />
          </joint>

        <gazebo reference="camera_rgb_frame">
            <sensor type="depth" name="camera">
              <always_on>true</always_on>
              <visualize>false</visualize>
              <update_rate>15.0</update_rate>
              <camera name="front">
                <horizontal_fov>1.047197</horizontal_fov>
                <image>
                  <format>B8G8R8</format>
                  <width>1920</width>
                  <height>1080</height>
                </image>
                <clip>
                  <near>0.01</near>
                  <far>9</far>
                </clip>
              </camera>
              <plugin name="camera_controller" filename="libgazebo_ros_openni_kinect.so">
                <baseline>0.1</baseline>
                <alwaysOn>true</alwaysOn>
                <updateRate>15.0</updateRate>
                <cameraName>camera</cameraName>
                <imageTopicName>/camera/rgb/image_raw</imageTopicName>
                <cameraInfoTopicName>/camera/rgb/camera_info</cameraInfoTopicName>
                <depthImageTopicName>/camera/depth_registered/image_raw</depthImageTopicName>
                <depthImageCameraInfoTopicName>/camera/depth_registered/camera_info</depthImageCameraInfoTopicName>
                <pointCloudTopicName>/camera/depth_registered/points</pointCloudTopicName>
                <frameName>camera_rgb_optical_frame</frameName>
                <pointCloudCutoff>0.35</pointCloudCutoff>
                <pointCloudCutoffMax>4.5</pointCloudCutoffMax>
                <CxPrime>0</CxPrime>
                <Cx>900</Cx>
                <Cy>540</Cy>
                <focalLength>1.0</focalLength>
                <hackBaseline>0</hackBaseline>
              </plugin>
            </sensor>
          </gazebo>

 <!--       <gazebo reference="cubic">
	        <material>Gazebo/Yellow</material>
	</gazebo>
        <gazebo reference="cylinder">
		<material>Gazebo/WoodTile</material>
	</gazebo>
        <gazebo reference="fourPrism">
		<material>Gazebo/WoodTile</material>
	</gazebo>
        <gazebo reference="fourPrismoid">
		<material>Gazebo/WoodTile</material>
	</gazebo>
        <gazebo reference="fourPyramid">
		<material>Gazebo/WoodTile</material>
	</gazebo>
        <gazebo reference="halfSphere">
		<material>Gazebo/WoodTile</material>
	</gazebo>
        <gazebo reference="shortCylinder">
		<material>Gazebo/WoodTile</material>
	</gazebo>
        <gazebo reference="sixPrism">
		<material>Gazebo/WoodTile</material>
	</gazebo>
        <gazebo reference="threePrism">
		<material>Gazebo/WoodTile</material>
	</gazebo>
        <gazebo reference="threePyramid">
		<material>Gazebo/WoodTile</material>
	</gazebo> 

      <link name="cubic">
		<inertial>
			<origin xyz="0 0 0" rpy="0.2 0.3 0.4" />
			<mass value="0.07" />
			<inertia ixx="0.02" ixy="0.02" ixz="0.02" iyy="0.02" iyz="0.02" izz="0.02" />
		</inertial>
		<visual>
			<origin xyz="0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh scale="0.01 0.01 0.01" filename="$(find robot_grasp)/models/grasp_object/meshes/cubic.STL" />
			</geometry>
		</visual>
		<collision>
			<origin xyz="0 0 0" rpy="0 0 0" />
			<geometry>
				<mesh scale="0.01 0.01 0.01" filename="$(find robot_grasp)/models/grasp_object/meshes/cubic.STL" />
			</geometry>
		</collision>
     </link> 

	<joint name="basket_cubic" type="floating">
		<origin xyz="0.04 0 0.03" rpy="0 0 0" />
		<parent link="basket" />
		<child link="cubic" />
		<axis xyz="0 0 0" />
	</joint>  -->


</root>
